### DISCUSSION TITLE
The code mentioned in the section "Input with Date Picker Dialog" throws an error

### DISCUSSION THREAD

**Discussion by karansethia** [gpbl/react-day-picker#2139] [General]
<img width="740" alt="Screenshot 2024-05-18 at 1 06 44 PM" src="https://github.com/gpbl/react-day-picker/assets/83359232/c50185d3-22ba-4949-950c-0d6ee5661f8e">

link => https://react-day-picker.js.org/advanced-guides/input-fields
<br />
 re-writing the onSelect prop worked our for me
<br />

<DayPicker
initialFocus
mode="single"
selected={selectedDate}
 onSelect={(date) => handleSelect(date as Date)}
/>

**Im not sure if this is the correct way and if there is a correct way to pass handleSelect function to onSelect prop please, update the docs**

**Comment by gpbl** [comment]
Thanks, there was a bug indeed in the docs. See https://react-day-picker.js.org/advanced-guides/input-fields#input-with-inline-calendar

**Reply by karansethia** [reply]
Happy to help, also since shadcn is also using your package, I dont know what the formal way to request is but I would love to see docs updated with how to use this component with react-hook-form. 
I created a seperate component for date input, had to sync it with refs using forwardRef and the actual code of form would look like this

> const Form = () => {
  const {register, handleSubmit, setValue, watch} = useForm<FormData>({
    defaultValues: {
      name: "Karan Sethia",
      age: 23,
      birthdate: "01/01/2000",
    },
  });

 

>  const onSubmit = (data: FormData) => {
>     console.log(data);
>   };
> 
>   //we use this watch from RHF to track the date
>   const birthdate = watch("birthdate");
> 
>   return (
>     <form onSubmit={handleSubmit(onSubmit)}>
>       <div>
>         <label htmlFor="name">Name</label>
>         <input id="name" {...register("name")} placeholder="Enter your name" />
>       </div>
>       <div>
>         <label htmlFor="age">Age</label>
>         <input
>           type="number"
>           id="age"
>           {...register("age")}
>           placeholder="Enter your age"
>         />
>       </div>
>       <div>
>         <label htmlFor="birthdate">Birthdate</label>
>         <DateInput
>           value={birthdate}
>           onChange={(e) => setValue("birthdate", e.target.value)}
>           wrapperClass="my-2"
>           placeholder="MM/DD/YYYY"
>         />
>       </div>
>       <button type="submit">Submit</button>
>     </form>
>   );
> };

Now in the actuall app im working on there are 8 date inputs and having to declare a variable for each seperately is sort of hassle.
I would love to see the component work out of the box with RHF and maybe other libraries
